src/CBT.NuGet/Internal/NuGetPackageReferenceProjectParser.cs

```diff
@@ -1,85 +1,125 @@
-        public NuGetPackageReferenceProjectParser(CBTTaskLogHelper logger)
+        public NuGetPackageReferenceProjectParser(ISettings settings, CBTTaskLogHelper log)
+            : base(settings, log)
         {
-            _log = logger;
         }
-        public IEnumerable<PackageIdentityWithPath> GetPackages(string packagesPath, string packageConfigPath, PackageRestoreData packageRestoreData)
+
+        public override bool TryGetPackages(string packageConfigPath, PackageRestoreData packageRestoreData, out IEnumerable<PackageIdentityWithPath> packages)
         {
-            if (packageRestoreData?.RestoreProjectStyle != null && !packageRestoreData.RestoreProjectStyle.Equals("PackageReference", StringComparison.InvariantCultureIgnoreCase))
+            packages = null;
+
+            // This assumes that if it is a non packages.config or project.json being restored that it is a MSBuild project using the new PackageReference.
+            if (ProjectJsonPathUtilities.IsProjectConfig(packageConfigPath) || NuGetPackagesConfigParser.IsPackagesConfigFile(packageConfigPath))
             {
-                yield break;
+                return false;
             }
-            // This assumes that if it is a non packages.config or project.json being restored that it is a MSBuild project using the new PackageReference.  
-            if (ProjectJsonPathUtilities.IsProjectConfig(packageConfigPath) || packageConfigPath.EndsWith(Constants.PackageReferenceFile, StringComparison.OrdinalIgnoreCase))
+
+            if (packageRestoreData == null)
             {
-                yield break;
+                Log.LogWarning("Missing expected assets file directory.  This is typically because the flag generated at $(CBTModuleNuGetAssetsFlagFile) does not exist or is empty.  Ensure the GenerateModuleAssetFlagFile target is running. It may also be because the CBTModules.proj does not import CBT build.props in some fashion.");
+                return false;
             }
-            if (string.IsNullOrWhiteSpace(packageRestoreData?.RestoreOutputAbsolutePath))
+
+            if (String.Equals("Unknown", packageRestoreData?.RestoreProjectStyle, StringComparison.OrdinalIgnoreCase))
             {
-                _log.LogMessage(MessageImportance.Low, "Missing assets file directory.  If you are running NuGet prior to 4.x this is expected.  If you are running NuGet 4.x or higher then this is typically because the flag generated at $(CBTNuGetAssetsFlagFile) does not exist or is empty.  Ensure the GenerateNuGetAssetFlagFile target is running. It may also be because the project does not import cbt build.props in some fashion.  It may also be because NuGet failed to parse the project you may set the env NUGET_RESTORE_MSBUILD_ARGS to the value '/flp:v=diag;logfile=restore.log;append' and check the restore.log for more details. ");
-                yield break;
+                packages = Enumerable.Empty<PackageIdentityWithPath>();
+
+                return true;
```

Comment:
>true [](start = 23, length = 4)

false?

Reply:
and comment that this is ordering specific

---
In reply to: [144648337](https://github.com/CommonBuildToolset/CBT.Modules/pull/182#discussion_r144648337) [](ancestors = 144648337)

Reply:
Done